From 1a7d03cf118fc74255dacc8dbb7567cc42fd0765 Mon Sep 17 00:00:00 2001
From: Konrad Leszczynski <konrad.leszczynski@intel.com>
Date: Tue, 15 Dec 2015 11:26:52 +0000
Subject: [PATCH 01/12] BACKPORT: usb: gadget: f_midi: enable use of the index
 parameter

The soundcard index to use for the ALSA device creation is passed as a
parameter to f_midi_bind_config(), but is assigned to midi->index only
after the call to f_midi_register_card(midi). So no matter what is passed
to f_midi_bind_config(), the actual index for snd_card_new() is always 0.
This probably works ok if at the moment of f_midi's bind there are no
other snd_cards, but if there are, it is not possible to bind f_midi.

This patch moves the assignment to a place before the call to
f_midi_register_card(midi).

Change-Id: If99c4c03c5c7a01904edf0b7d5c50f21a2b0aa27
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-3355
Signed-off-by: Andrzej Pietrasiewicz <andrzej.p@samsung.com>
Signed-off-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: Konrad Leszczynski <konrad.leszczynski@intel.com>
Reviewed-on: https://android.intel.com:443/448826
---
 drivers/usb/gadget/f_midi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/f_midi.c b/drivers/usb/gadget/f_midi.c
index 36d4bb2..4b8af19 100644
--- a/drivers/usb/gadget/f_midi.c
+++ b/drivers/usb/gadget/f_midi.c
@@ -955,6 +955,7 @@ int __init f_midi_bind_config(struct usb_configuration *c,
 	/* set up ALSA midi devices */
 	midi->in_ports = in_ports;
 	midi->out_ports = out_ports;
+	midi->index = index;
 	status = f_midi_register_card(midi);
 	if (status < 0)
 		goto setup_fail;
@@ -967,7 +968,6 @@ int __init f_midi_bind_config(struct usb_configuration *c,
 	midi->func.disable     = f_midi_disable;
 
 	midi->id = kstrdup(id, GFP_KERNEL);
-	midi->index = index;
 	midi->buflen = buflen;
 	midi->qlen = qlen;
 
-- 
1.9.1

